src/powershell/Add-ExportScopeCredential.psm1

```diff
@@ -0,0 +1,109 @@
+<#
+.SYNOPSIS
+Creates secrets in Azure KeyVault for the Service Principal.
+
+.PARAMETER HubName
+The name of the FinOps Hub.  
+Assumes HubName = ResourceGroupName.
+
+.PARAMETER ClientId
+The Application ID (Client ID) of the Service Principal to create the secret for.
+
+.PARAMETER TenantId
+The ID of the Azure AD Tenant ID containing the Service Principal.
+
+.PARAMETER Secret
+The Secret (Password) of the service principal.
+
+.EXAMPLE
+$clientId = ConvertTo-SecureString -AsPlainText -String '00000000-0000-0000-0000-000000000000'
+$secret = ConvertTo-SecureString -AsPlainText -String 'SomePassword'
+Add-ExportScopeCredential -HubName FinOps-Hub  -TenantId $tenantid -ClientId $clientId -Secret $secret
+
+Creates secrets in Azure KeyVault for the Service Principal.
+#>
+function Add-ExportScopeCredential {
+    [CmdletBinding()]
+    param (
+        [Parameter()]
+        [string]
+        [ValidateNotNullOrEmpty()]
+        $HubName,
+        [Parameter()]
+        [String]
+        [ValidateNotNullOrEmpty()]
+        $TenantId,
+        [Parameter()]
+        [Securestring]
+        [ValidateNotNullOrEmpty()]
+        $ClientId,
+        [Parameter()]
+        [Securestring]
+        [ValidateNotNullOrEmpty()]
+        $Secret
+    )
+    $ErrorActionPreference = 'Stop'
+    [string]$ClientIdSecretName = 'clientId{0}' -f $TenantId
+    [string]$SecretSecretName = 'secret{0}' -f $TenantId
+
+    # Functions
+    function Set-secret {
```

Comment:
Do we need to add anything to the secret so we know what scope it maps back to?

Reply:
The secrets have the TenantId as part of the name - secret<tenantId> and clientId<tenantId>.
The TenantId is stored as part of the exportScope so we know which credentials to use for the scope.